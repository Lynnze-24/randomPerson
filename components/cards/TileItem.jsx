import React from 'react'
import { View, Text,StyleSheet } from 'react-native'
import { Avatar } from 'react-native-paper'
import useMediaQueries from '../../hooks/MediaQueries'
import cusTheme from '../../theme/Theme'
import IconText from '../IconText'

const TileItem = ({name,avatar,email,phone,location}) => {
    const {isLessThanLG,isLessThanSM}= useMediaQueries()
    return (
        <View testID='tileItem' style={[styles.container,
                     isLessThanLG && styles.tileLG,
                      isLessThanSM && styles.tileSM,
                    
                     ]}>
            
            <Avatar.Image testID='avatar' size={60} source={avatar} />
            
           
            <View style={{width:250}}>
            <Text style={styles.text}>{name}</Text>
            <IconText isTable={false} icon="email-outline" text={email} style={{flex:1.5}}/>
            <IconText isTable={false} icon="phone-outline" text={phone} />
            <IconText isTable={false} type="Ionicons" icon="location" text={location} style={{flex:1.5}}/>
            </View>
        </View>
    )
}

const styles = StyleSheet.create({
    container: {
      height:125,
      width:400,
      backgroundColor:cusTheme.secondary,
      margin:10,
      borderRadius:10,
      justifyContent:'space-evenly',
      flexDirection:'row',
      alignItems:'center',
      shadowColor: "#000",
      shadowOffset: {
      width: 0,
      height: 5,
     },
     shadowOpacity: 0.36,
     shadowRadius: 6.68,

     elevation: 11,
   },
   tileLG:{
       height:140,
      width:530,
   },
   tileSM:{
    height:120,
    width:380,
    },
   text:{
       width:200,
       fontSize:18
   }
})

export default TileItem;
