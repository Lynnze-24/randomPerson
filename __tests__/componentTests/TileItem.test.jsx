import {render, fireEvent } from "@testing-library/react-native";
import TileItem from "../../components/cards/TileItem.jsx"
import renderer from 'react-test-renderer';



it('matches snapShot', () => {
    const comp = renderer.create(<TileItem name="Aung"
                                    avatar={1} // mock
                                    email="gmail.com"
                                    phone="999"
                                    location="bago"
                                    />).toJSON();
    expect(comp).toMatchSnapshot();
});



it('should render component correctly',()=>{ 
    
    const {getByText,getByTestId}=render(
                        <TileItem name="Aung"
                                   avatar={1} // mock
                                   email="gmail.com"
                                   phone="999"
                                   location="bago"
                                    />
                        );
            getByTestId('avatar')
            getByText("Aung")
            getByText("gmail.com")
            getByText("999")
            getByText("bago")
        
})
