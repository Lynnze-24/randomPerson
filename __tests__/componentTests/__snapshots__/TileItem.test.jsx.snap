// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapShot 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#f0edf6",
        "borderRadius": 10,
        "elevation": 11,
        "flexDirection": "row",
        "height": 125,
        "justifyContent": "space-evenly",
        "margin": 10,
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 5,
          "width": 0,
        },
        "shadowOpacity": 0.36,
        "shadowRadius": 6.68,
        "width": 400,
      },
      Object {
        "height": 140,
        "width": 530,
      },
      false,
    ]
  }
  testID="tileItem"
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#6200ee",
          "borderRadius": 30,
          "height": 60,
          "width": 60,
        },
        undefined,
      ]
    }
    testID="avatar"
  >
    <Image
      source={1}
      style={
        Object {
          "borderRadius": 30,
          "height": 60,
          "width": 60,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "width": 250,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 18,
          "width": 200,
        }
      }
    >
      Aung
    </Text>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
          },
          Object {
            "flex": 1.5,
          },
        ]
      }
    >
      <Text />
      <Text
        style={
          Object {
            "fontSize": 13,
            "marginLeft": 2,
            "textAlign": "left",
          }
        }
      >
        gmail.com
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
          },
          undefined,
        ]
      }
    >
      <Text />
      <Text
        style={
          Object {
            "fontSize": 13,
            "marginLeft": 2,
            "textAlign": "left",
          }
        }
      >
        999
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
          },
          Object {
            "flex": 1.5,
          },
        ]
      }
    >
      <Text />
      <Text
        style={
          Object {
            "fontSize": 13,
            "marginLeft": 2,
            "textAlign": "left",
          }
        }
      >
        bago
      </Text>
    </View>
  </View>
</View>
`;
